finish/src/main/java/io/openliberty/guides/inventory/model/InventoryList.java

```diff
@@ -32,45 +34,14 @@ public void addToInventoryList(String hostname, Properties systemProps) {
         Properties props = new Properties();
         props.setProperty("os.name", systemProps.getProperty("os.name"));
         props.setProperty("user.name", systemProps.getProperty("user.name"));
-
-        SystemEntry host = new SystemEntry(hostname, props);
-        if (!systems.contains(host))
-            systems.add(host);
+        systems.put(hostname, props);
     }
 
     public Properties findHost(String hostname) {
-        for (SystemEntry system : systems) {
-            if (system.getHostname().equals(hostname)) {
-                return system.getProperties();
-            }
-        }
-        return null;
+    	if (systems.containsKey(hostname)){
+    		return systems.get(hostname);
```

Comment:
too much indentation, you should uses space for indentation instead of tabs here. Also `systems.get()` should be enough as `get` returns null if no key:value mapping exists in the hashmap: https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#get-java.lang.Object-